packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
+          <span class="active-option-label">{this.active?.label ?? "Selecteer een optie"}</span>
+          <span class="active-option-aside">
+            {this.options.some((option) => "summaryCounter" in option && option?.summaryCounter) ? (
+              <span class="badges">
+                {this.options
+                  .filter((option) => "summaryCounter" in option && option?.summaryCounter)
+                  .map((option) => {
+                    if ("options" in option) {
+                      return <dso-badge status={option.variant ?? "outline"}>{option.options.length}</dso-badge>;
+                    }
+                  })}
+              </span>
+            ) : (
+              ""
+            )}
+            <dso-icon icon="caret-down"></dso-icon>
+          </span>
+        </button>
+        <div class={{ "groups-container": true, open: this.open }}>
+          <ul class="groups">
+            {this.options.map((optionsOrGroup) => {
+              if ("options" in optionsOrGroup) {
+                return (
+                  <li class={{ group: true, [`group-${optionsOrGroup.variant}`]: !!optionsOrGroup.variant }}>
+                    {optionsOrGroup.label && <p class="group-label">{optionsOrGroup.label}</p>}
+                    <ul class="options">
+                      {optionsOrGroup.options.map((option) => (
+                        <OptionElement
+                          option={option}
+                          active={this.active}
+                          activeHint={this.activeHint}
+                          cb={this.handleOptionClick}
+                        />
+                      ))}
+                    </ul>
+                    {optionsOrGroup.redirect && (
+                      <a
+                        class="group-link"
+                        href={optionsOrGroup.redirect.href}
+                        onClick={(e: MouseEvent) =>
+                          this.handleRedirectClick(e, optionsOrGroup.redirect as AdvancedSelectGroupRedirect)
```

Comment:
Vanwaar de type assertion?

Reply:
Typescript herkent de `{optionsOrGroup.redirect && (` check niet en denkt nog steeds dat het `undefined` kan zijn.
![image](https://github.com/dso-toolkit/dso-toolkit/assets/6705457/13731498-4481-4739-8b61-9cc27a7532ef)

Reply:
Dat komt omdat de `onClick` event handler asynchroon is: Je zal opnieuw moeten testen of het object wat je meegeeft ook echt is wat je denkt dat het is.

Reply:
Logisch, maar stiekem toch jammer 😜 Heb het toegevoegd.